spec-0010/index.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "SPEC 10 — Changelog"
+number: 10
+date: 2024-06-03
+author:
+  - "Inessa Pawson <inessapawson@gmail.com>"
+discussion: https://discuss.scientific-python.org/t/
+endorsed-by:
+---
+
+## Description
+
+SPEC 10 provides guidelines and best practices for maintaining a changelog file
+for libraries in the Scientific Python ecosystem.
+
+### Core Project Endorsement
+
+<!--
+Briefly discuss what it means for a core project to endorse this SPEC.
+-->
+
+### Ecosystem Adoption
+
+The endorsement of this SPEC signifies your project's support for the guidelines 
+laid out in the document.
+
+## Implementation
+
+Keeping a clear, organized, and human-readable record of notable changes for each 
+version of a Python library is essential for the project’s maintenance and sustainability. 
+It promotes transparency and trust between the project’s leadership team, developers, and users.
+
+### What Is a Changelog?
+
+A changelog is a file which contains a curated, chronologically ordered list of 
+notable changes for each version of a project.
+
+### Guiding Principles
```

Comment:
Changelogs are not meant to take up a bunch of developer time, slowing down releases.

Reply:
That statement always sounded a bit weird to me because I consider release notes / changelog to be documentation. It's a bit like saying "examples are not meant to take up a bunch of developer time...".

Reply:
But I agree, making it as easy as possible to achieve that outcome is why we have tools like towncrier / changelist. :D

Reply:
I think it's more about avoiding busywork; i.e., write the descriptions when the PR is being reviewed, so you distribute the effort a bit, and then automate the release process as much as possible. Releases are very difficult to get out, so if it becomes as simple as tag-and-release, we get our tools in the hands of the users more expediently.